metabase/metabase
# Add `prettify-native-form` driver multimethod and refactor `mdb.query/format-sql`

Closes #34235

### Description

This PR extends driver with new multimethod, `driver/prettify-native-form`, which is used eg. to pretty format output of `dataset/native` endpoint. Also `mdb.query/format-sql` was refactored to use this method.

Even if we do not currently need to format nosql native code, I've decided to put the method in driver namespace, so it could be used as identity with nosql native code. This way we can avoid branchings or `ignore-exceptions` calls as previously in `/dataset/native` and underlying formatting logic. 

With this method in place, external driver developers can plug in their own formatters or reuse our formatting logic. To find out how, read `driver/prettify-native-form`'s docstring.

### How to verify

Multiple tests were modified during the refactor. Run those and see the results.

### Checklist
